### PR TITLE
Modify rule S2065: improve rule in the scope of LayC

### PR DESCRIPTION
**Pull request by johann-beleites-sonarsource**

### CODE REVIEW

**Comment by johann-beleites-sonarsource** [comment]
Old: https://next.sonarqube.com/sonarqube/coding_rules?q=S2065&open=java%3AS2065
New:
image
image
image

File: rules/S2065/java/rule.adoc [hunk 1]
```diff
@@ -1,29 +1,50 @@
...
+Fields marked as `transient` in a `Serializable` class will be ignored during serialization and consequently not written out to a file (or stream). 
 
+This can be useful in situations such as where the content of a field can be recomputed from other fields.
+To reduce the output size, this field can be marked as `transient` and recomputed when a given object is deserialized.
 
-=== Noncompliant code example
+Since `transient` is very specific to classes that implement `Serializable`, it is superfluous in classes that do not.
 
-[source,java]
+This rule raises an issue when a field is marked as `transient`, even though the containing class is not marked as `Serializable`.
```
Comment (alban-auzeill) [inline rules/S2065/java/rule.adoc:10]:
```suggestion
This rule raises an issue when a field is marked as `transient`, even though the containing class does not implement `Serializable`.
```

File: rules/S2065/java/rule.adoc [hunk 2]
```diff
@@ -1,29 +1,50 @@
...
+==== Compliant solution
 
-=== Compliant solution
-
-[source,java]
+[source,java,diff-id=1,diff-type=compliant]
 ----
 class Vegetable {
-  private Season ripe;
+  private Season ripe; // Compliant, the field is not marked as "transient" modifier
```
Comment (alban-auzeill) [inline rules/S2065/java/rule.adoc:37]:
```suggestion
  private Season ripe; // Compliant, the field is not marked as "transient"
```

**Review by alban-auzeill — APPROVED** [review]
see my 2 suggestions

**Pull request merged**